### PR TITLE
Add a lucene test that saves all possible unicode characters

### PR DESCRIPTION
**Pull request by ScottDugas**
Just to make sure it doesn't break

### CODE REVIEW

File: fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java [hunk 1]
```diff
@@ -425,6 +429,74 @@ void testQueryWithStopWords() {
...
+                        .setGroup(1)
+                        .setText(text)
+                        .build();
+                recordStore.saveRecord(document1);
+                TestRecordsTextProto.SimpleDocument document2 = TestRecordsTextProto.SimpleDocument.newBuilder()
+                        .setDocId(id++)
+                        .setGroup(1)
+                        .setText(beginningWord + " Good morning the Mr Tian, would you like a " + endWord)
+                        .build();
+                recordStore.saveRecord(document2);
```
Comment (qdnguyen0) [inline fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java:485]: What's the purpose of `document2`?

Comment (ScottDugas) [reply fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java]: I think it was just a side effect of copying the test, and I didn't clean it up. It's removed now.

**Review by qdnguyen0 — APPROVED** [review]

**Comment by ScottDugas** [comment]
```
FDBOrQueryToUnionTest > testOrQuery5WithLimits(int, OrQueryParams) > com.apple.foundationdb.record.provider.foundationdb.query.FDBOrQueryToUnionTest.testOrQuery5WithLimits(int, OrQueryParams)[36] FAILED
    java.util.NoSuchElementException
        at java.base/java.util.ArrayDeque.removeFirst(ArrayDeque.java:363)
        at java.base/java.util.ArrayDeque.remove(ArrayDeque.java:524)
        at com.apple.foundationdb.record.cursors.MapPipelinedCursor.close(MapPipelinedCursor.java:94)
        at com.apple.foundationdb.record.cursors.MapResultCursor.close(MapResultCursor.java:80)
        at com.apple.foundationdb.record.cursors.MapResultCursor.close(MapResultCursor.java:80)
        at com.apple.foundationdb.record.provider.foundationdb.cursors.MergeCursorState.close(MergeCursorState.java:99)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at com.apple.foundationdb.record.provider.foundationdb.cursors.MergeCursor.close(MergeCursor.java:314)
        at com.apple.foundationdb.record.cursors.FilterCursor.close(FilterCursor.java:69)
        at com.apple.foundationdb.record.cursors.RowLimitedCursor.close(RowLimitedCursor.java:80)
        at com.apple.foundationdb.record.cursors.MapResultCursor.close(MapResultCursor.java:80)
        at com.apple.foundationdb.record.RecordCursorIterator.close(RecordCursorIterator.java:149)
        at com.apple.foundationdb.record.provider.foundationdb.query.FDBOrQueryToUnionTest.testOrQuery5WithLimits(FDBOrQueryToUnionTest.java:831)
```

**Review by qdnguyen0 — APPROVED** [review]
